src/dominators.rs

```diff
@@ -0,0 +1,277 @@
+//! # The Dominance Relation
+//!
+//! In a directed graph with a root node `R`, a node `A` is said to *dominate* a
+//! node `B` iff every path from `R` to `B` contains `A`.
+//!
+//! The node `A` is daid to *strictly dominate the node `B` iff `A` dominates
+//! `B` and `A != B`.
+//!
+//! The node `A` is said to be the *immediate dominator* of a node `B` iff it
+//! strictly dominates `B` and does not dominate any other nodes which also
+//! dominate `B` in turn.
+
+use std::collections::{HashMap, HashSet};
+use std::hash::Hash;
+
+use visit::{DfsPostOrder, GraphBase, IntoNeighbors, Visitable, Walker};
+
+/// The dominance relation for some graph and root.
+#[derive(Debug, Clone)]
+pub struct Dominators<G>
```

Comment:
I think this should be `Dominators<N>` (where `N` is the `G::NodeId`).

The reason is that the `G` as used in this file is a *reference-to-graph* (due to how lifetimes + iterators + traits are limited in Rust, that's what we use).  We don't want to tie the Dominators struct to a borrow of a graph, when no borrowed data is actually needed.